crates/agglayer-certificate-orchestrator/src/tests/certifier_results.rs

```diff
@@ -308,31 +331,81 @@ async fn certifier_error_proof_already_exists_unknown_network_at_height_0_but_no
     assert!(receiver.try_recv().is_err());
 }
 
+enum ExpectedAction {
+    GetCertificateHeader(CertificateStatus),
+    RemovePendingCertificate,
+    RemoveGeneratedProof,
+    UpdateCertificateHeaderStatus(CertificateStatus),
+    SetLatestProvenCertificatePerNetwork,
+    SpawnNextTask,
+}
 #[rstest]
-#[case::unknown_network_height_0_no_header(0, CertificateStatus::Candidate, false)]
-#[case::unknown_network_height_0_candidate(0, CertificateStatus::Candidate, false)]
-#[case::unknown_network_height_0_pending(0, CertificateStatus::Pending, false)]
-#[case::unknown_network_height_0_proven(0, CertificateStatus::Proven, false)]
-#[case::unknown_network_height_0_settled(0, CertificateStatus::Settled, false)]
-#[case::unknown_network_height_0_in_error(0, CertificateStatus::InError {
-    error: CertificateStatusError::TypeConversionError(
-        agglayer_types::Error::MultipleL1InfoRoot,
-)}, false)]
-#[case::known_network_height_not_0_candidate(1, CertificateStatus::Candidate, true)]
-#[case::known_network_height_not_0_pending(1, CertificateStatus::Pending, true)]
-#[case::known_network_height_not_0_proven(1, CertificateStatus::Proven, true)]
-#[case::known_network_height_not_0_settled(1, CertificateStatus::Settled, true)]
-#[case::known_network_height_not_0_in_error(1, CertificateStatus::InError {
-    error: CertificateStatusError::TypeConversionError(
-        agglayer_types::Error::MultipleL1InfoRoot,
-)}, true)]
+// The state doesn't know the network and the height is not 0, we remove the proof.
+#[case::unknown_network_height_not_0(1, None, vec![ExpectedAction::RemoveGeneratedProof])]
```

Comment:
When we receive a certificate for a network that didn't commit anything yet

Reply:
in that case we wouldn't have triggered the proving request to the agglayer prover in the first place no?

Reply:
That's an outside expectation that we don't trigger the proving under those condition, but it can technically happen in the different branch and needs to be covered. We have logs that will alert us if it ever happens in production.